### ISSUE TITLE
训练loss从十几或者二十几轮开始变为nan，并且dice也在逐渐下降

### ISSUE THREAD

**Issue by Xingchao003** [tiffany9056/UU-Mamba#5]
您好，我在训练时按照您所给的配置进行训练，但是在训练十几或者二十几个epochs后，损失值就突然变成nan并且dice指数也在逐渐下降，请问您知道是什么原因吗？

![Image](https://github.com/user-attachments/assets/37c9cee8-7dc3-48d6-93d2-c31409306811)

**Comment by tiffany9056** [comment]
您好，我没有遇到这个情况，但是您可以尝试调整您的learning rate看是否会有差

**Comment by Mr-wan123** [comment]
我在复现的过程中也遇到了这个问题，调整了学习率和SAM的参数还是不行，不知道现在有没有什么解决方法呢Duplicate of #

**Comment by renlyly** [comment]
Hi, thank you for your work!

I'm currently trying to reproduce your results, but I ran into the same issue mentioned above. I’ve tried adjusting the learning rate and the SAM parameters, but the problem persists.

Also, I encountered compatibility issues when installing uumamba with default PyTorch 2.0.1, so I had to upgrade to 2.1.2. Which version of Mamba-SSM is compatible or recommended?

Additionally, I noticed that uumamba refers to nnU-Net version 2.3.2, but I couldn’t find this version on the official nnU-Net GitHub repository. In order to run the processing and training step,  an older version of nnU-Net is needed to run it successfully. I am not sure if this is the reason

Thanks again for your contribution!

**Comment by leekak** [comment]
Hi, I encountered the same NaN issue while reproducing UU-Mamba on ACDC.

In my case, the training became NaN after several epochs when using the default fp16 AMP setting. Lowering the learning rate alone did not solve it.

What fixed it for me was switching AMP from fp16 to bf16 and making `UMambaEnc.py` handle `torch.bfloat16` correctly.

In `nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py`, I changed the CUDA autocast contexts to use bfloat16:

    with autocast(self.device.type, enabled=True, dtype=torch.bfloat16) if self.device.type == 'cuda' else dummy_context():

Then, in `nnunetv2/nets/UMambaEnc.py`, the original code only converted fp16 inputs to float32:

    if x.dtype == torch.float16:
        x = x.type(torch.float32)

I changed it to:

    if x.dtype in (torch.float16, torch.bfloat16):
        x = x.type(torch.float32)

After these two changes, the NaN issue disappeared in my reproduction.

Hope this helps.
